driver-core/src/main/com/mongodb/internal/connection/OidcAuthenticator.java

```diff
@@ -206,6 +217,26 @@ public void authenticate(final InternalConnection connection, final ConnectionDe
         }
     }
 
+    @Override
+    void authenticateAsync(
+            final InternalConnection connection,
+            final ConnectionDescription connectionDescription,
+            final SingleResultCallback<Void> callback) {
+        assertFalse(connection.opened());
```

Comment:
Let's either add the two comments that exist in `authenticate`, or remove both. I think that the first comment (the one duplicating assertion) is not useful, while the second one may be turned in the method's API doc.

Reply:
It seems the second one is impl, so does not fit as an API doc.